source/simulator/core.cc

```diff
@@ -2241,6 +2256,297 @@ namespace aspect
+                            }
+                        }
+                      else
+                        {
+                          if (0.9*parameters.linear_stokes_solver_tolerance * parameters.linear_stokes_solver_tolerance <= 0.1)
+                            {
+                              parameters.linear_stokes_solver_tolerance =  std::min(parameters.minimum_linear_stokes_solver_tolerance, 0.9 * std::fabs(newton_residual*newton_residual)/(newton_residual_old*newton_residual_old));
+                            }
+                          else
+                            {
+                              parameters.linear_stokes_solver_tolerance =  std::min(parameters.minimum_linear_stokes_solver_tolerance,
+                                                                                    std::max(0.9 * std::fabs(newton_residual*newton_residual)/(newton_residual_old*newton_residual_old),0.9*parameters.linear_stokes_solver_tolerance*parameters.linear_stokes_solver_tolerance));
+                            }
+                        }
+                      pcout << std::endl << "   The linear solver tolerance is set to " << parameters.linear_stokes_solver_tolerance << ":" << std::fabs(newton_residual-stokes_residual)/newton_residual_old << "," << stokes_residual << "," << newton_residual << ", " << newton_residual_old << std::endl;
+
+                    }
+                }
+              build_stokes_preconditioner();
+
+
+              stokes_residual = solve_stokes();
+
+              /* The following block may be useful for when the newton solver fails. Is this somewthing we want?
+                            try
+                            {
+                                stokes_residual = solve_stokes();
+                            }
+                            catch (...)
+                            {
+                              // first get out of the timing section
+                              computing_timer.exit_section();
+                              // Newton solver failed. Try Picard instead.
+                              use_picard = true;
+                              parameters.newton_theta = 0;
+                                rebuild_stokes_matrix = rebuild_stokes_preconditioner = assemble_newton_stokes_matrix = true;
+                                assemble_stokes_system();
+                                parameters.linear_stokes_solver_tolerance = piccard_linear_stokes_solver_tolerance;
+                                pcout << std::endl << "   Solve failed, set Use Picard to true. and the linear solver tolerance is set to "
+                                  << parameters.linear_stokes_solver_tolerance << std::endl;
+                                stokes_residual = solve_stokes();
+                            }*/
```

Comment:
Remove it, and leave this for a separate pull request, we will also need a test that it works.